actionpack/lib/abstract_controller/base.rb

```diff
@@ -88,14 +88,13 @@ def internal_methods
       # * <tt>Set</tt> - A set of all methods that should be considered actions.
       def action_methods
         @action_methods ||= begin
+
           # All public instance methods of this class, including ancestors
           methods = (public_instance_methods(true) -
             # Except for public instance methods of Base and its ancestors
             internal_methods +
             # Be sure to include shadowed public instance methods of this class
-            public_instance_methods(false))
-
-          methods.map!(&:to_s)
+            public_instance_methods(false)).map(&:to_s)
```

Comment:
This regresses a performance optimisation from https://github.com/rails/rails/pull/33693.

```suggestion
            public_instance_methods(false))
            
          methods.map!(&:to_s)
```

Reply:
Well caught, thank you